lasagne/objectives.py

```diff
@@ -338,6 +338,50 @@ def multiclass_hinge_loss(predictions, targets, delta=1):
     return theano.tensor.nnet.relu(rest - corrects + delta)
 
 
+def huber_loss(predictions, targets, delta=1):
+    """ Computes the huber loss between predictions and targets.
+
+    .. math:: L_\delta (diff) = \frac{diff^2}{2} & \text{if  } |diff| \le \
+                                \delta
+    .. math:: L_\delta (diff) & = & \delta (|diff| - \frac{\delta}{2} ),\
+                                     &\text{else}
+
+    Parameters
+    ----------
+    predictions : Theano 2D tensor
+        Predictions in (0, 1), such as softmax output of a neural network,
+        with data points in rows and class probabilities in columns.
+    targets : Theano 2D tensor or 1D tensor
+        Either a vector of int giving the correct class index per data point
+        or a 2D tensor of one-hot encoding of the correct class in the same
+        layout as predictions (non-binary targets in [0, 1] do not work!)
+    delta : scalar, default 1
+        This delta value is defaulted to 1, for SmoothL1Loss
+        described in Fast-RCNN paper[1].
+
+    Returns
+    -------
+    Theano 1D tensor
+        An expression for the item-wise huber loss.
```

Comment:
Again, this was copied from a multi-class objective docstring. Copy it from the squared error instead (it should be a tensor of arbitrary dimensionality, and the element-wise loss).